packages/grid/x-data-grid-pro/src/DataGridPro.tsx

```diff
@@ -359,26 +359,26 @@ DataGridProRaw.propTypes = {
   /**
    * Callback fired when the cell changes are committed.
    * @param {GridCellEditCommitParams} params With all properties from [[GridCellEditCommitParams]].
-   * @param {MuiEvent<React.SyntheticEvent>} event The event that caused this prop to be called.
+   * @param {MuiEvent<MuiBaseEvent>} event The event that caused this prop to be called.
    * @param {GridCallbackDetails} details Additional details for this callback.
    */
   onCellEditCommit: PropTypes.func,
   /**
    * Callback fired when the cell turns to edit mode.
    * @param {GridCellParams} params With all properties from [[GridCellParams]].
-   * @param {MuiEvent<React.SyntheticEvent>} event The event that caused this prop to be called.
+   * @param {MuiEvent<React.KeyboardEvent | React.MouseEvent>} event The event that caused this prop to be called.
    */
   onCellEditStart: PropTypes.func,
   /**
    * Callback fired when the cell turns to view mode.
    * @param {GridCellParams} params With all properties from [[GridCellParams]].
-   * @param {MuiEvent<React.SyntheticEvent>} event The event that caused this prop to be called.
+   * @param {MuiEvent<MuiBaseEvent>} event The event that caused this prop to be called.
```

Comment:
Why `MuiEvent<MuiBaseEvent>` here but `MuiEvent<React.KeyboardEvent | React.MouseEvent>` in the previous one?

I would prefer to be explicit in the generic than to add another layer of complexity.

Reply:
Because in `commitPropsAndExit` you receive a `MuiBaseEvent` and you publish a `cellEditStop` with it.
It is a `MuiBaseEvent` there because the `cellFocusOut` passes a `MuiBaseEvent` when triggered by `setColumnHeaderFocus`.

I did not wanted to change the signature of `setColumnHeaderFocus` but I can if it makes sense.

What I did for those events is to find the most restrictive set of events that matches all our calls.